### PR TITLE
PIL-748 update - Contact Email and telephone

### PR DESCRIPTION
**Pull request by javed-iqbal1**

### CODE REVIEW

**Review by aashik-varahan — DISMISSED** [review]

**Comment by jassalrichy** [comment]
Please update the regex and place in forms.Validation object:
val telephoneRegex = """^\+[0-9 ]{1,18}$|^[0-9 ]{1,19}$"""

**Review by jassalrichy — CHANGES_REQUESTED** [review]
Please update the regex and place in forms.Validation object:
val telephoneRegex = """^+[0-9 ]{1,18}$|^[0-9 ]{1,19}$"""

File: app/forms/RfmCaptureTelephoneDetailsFormProvider.scala [hunk 1]
```diff
@@ -23,14 +23,13 @@ import javax.inject.Inject
...
   private val phoneNumberLength = 24
-  private val phoneRegex        = """^[A-Z0-9 )/(\-*#+]*$"""
   def apply(userName: String): Form[String] = Form(
     "telephoneNumber" ->
       text("rfmCaptureTelephoneDetails.error.required", Seq(userName))
         .verifying(
           firstError(
             maxLength(phoneNumberLength, "rfmCaptureTelephoneDetails.messages.error.length"),
-            regexp(phoneRegex, "rfmCaptureTelephoneDetails.messages.error.format")
+            regexp(forms.Validation.telephoneRegex, "rfmCaptureTelephoneDetails.messages.error.format")
```
Comment (jassalrichy) [inline app/forms/RfmCaptureTelephoneDetailsFormProvider.scala:32]: Will this work? Do we not need to import forms.Validation

File: app/forms/RfmCaptureTelephoneDetailsFormProvider.scala [hunk 2]
```diff
@@ -16,21 +16,21 @@
...
 package forms
 
+import forms.Validation.telephoneRegex
 import forms.mappings.Mappings
 import play.api.data.Form
 
 import javax.inject.Inject
 
 class RfmCaptureTelephoneDetailsFormProvider @Inject() extends Mappings {
   private val phoneNumberLength = 24
```
Comment (jassalrichy) [inline app/forms/RfmCaptureTelephoneDetailsFormProvider.scala:26]: Sorry, can we add this to the mapping.Constant object:
final val TELEPHONE_NUMBER_MAX_LENGTH = 24

File: app/forms/Validation.scala [hunk 3]
```diff
@@ -21,4 +21,6 @@ object Validation {
   final val EMAIL_REGEX =
     """^(?!\.)("([^"\r\\]|\\["\r\\])*"|([-a-zA-Z0-9!#$%&'*+/=?^_`{|}~]|(?<!\.)\.)*)(?<!\.)@[a-zA-Z0-9][\w\.-]*[a-zA-Z0-9]\.[a-zA-Z][a-zA-Z\.]*[a-zA-Z]$"""
   final val GROUPID_REGEX = "^X[A-Z]PLR[0-9]{10}$"
+  val telephoneRegex      = "^[0-9 +()]{0,25}$"
```
Comment (jassalrichy) [inline app/forms/Validation.scala:24]: make this **final val**

**Review by jassalrichy — APPROVED** [review]

**Pull request merged**